librosa/segment.py

```diff
@@ -38,13 +39,213 @@
 from . import util
 from .util.exceptions import ParameterError
 
-__all__ = ['recurrence_matrix',
+__all__ = ['cross_similarity',
+           'recurrence_matrix',
            'recurrence_to_lag',
            'lag_to_recurrence',
            'timelag_filter',
            'agglomerative',
            'subsegment']
 
+@cache(level=30)
+def cross_similarity(data1, data2, k=None, width=1, metric='euclidean',
+                     sparse=False, mode='connectivity', bandwidth=None):
+    '''Compute cross similarity between two data sequences.
+
+
+    `xsim[i, j]` is non-zero if (`data1[:, i]`, `data2[:, j]`) are
+    k-nearest-neighbors and `|i - j| >= width`
+
+
+    Parameters
+    ----------
+    data1 : np.ndarray [shape=(K, N)]
+        A feature matrix for sequence 1
+
+    data2 : np.ndarray [shape=(K, M)]
+        A feature matrix for sequence 2
```

Comment:
it would be better to give these parameters more descriptive names, eg `data_to` and `data_from`